packages/core/src/app/billing/Billing.tsx

```diff
@@ -1,100 +1,64 @@
-    googleMapsApiKey: string;
-    isInitializing: boolean;
-    isUpdating: boolean;
-    shouldShowOrderComments: boolean;
-    billingAddress?: Address;
-    methodId?: string;
-    isFloatingLabelEnabled?: boolean;
-    themeV2?: boolean;
-    getFields(countryCode?: string): FormField[];
-    initialize(): Promise<CheckoutSelectors>;
-    updateAddress(address: Partial<Address>): Promise<CheckoutSelectors>;
-    updateCheckout(payload: CheckoutRequestBody): Promise<CheckoutSelectors>;
-}
-
-class Billing extends Component<BillingProps & WithCheckoutBillingProps> {
-    async componentDidMount(): Promise<void> {
-        const { initialize, onReady = noop, onUnhandledError } = this.props;
+const Billing = ({navigateNextStep, onReady, onUnhandledError}:BillingProps): ReactElement => {
+    const { checkoutService, checkoutState } = useCheckout();
 
-        try {
-            await initialize();
-            onReady();
-        } catch (error) {
-            if (error instanceof Error) {
-                onUnhandledError(error);
-            }
-        }
-    }
+    const {
+        data: {
+            getCheckout,
+            getConfig,
+            getCart,
+            getCustomer,
+            getBillingAddress,
+            getBillingAddressFields,
+        },
+        statuses: { isLoadingBillingCountries },
+    } = checkoutState;
+    const config = getConfig();
+    const customer = getCustomer();
+    const checkout = getCheckout();
+    const cart = getCart();
 
-    render(): ReactNode {
-        const { updateAddress, isInitializing, themeV2, ...props } = this.props;
-
-        return (
-            <AddressFormSkeleton isLoading={isInitializing}>
-                <div className="checkout-form">
-                    <div className="form-legend-container">
-                        <Legend testId="billing-address-heading" themeV2={themeV2}>
-                            <TranslatedString id="billing.billing_address_heading" />
-                        </Legend>
-                    </div>
-                    <BillingForm
-                        {...props}
-                        onSubmit={this.handleSubmit}
-                        updateAddress={updateAddress}
-                    />
-                </div>
-            </AddressFormSkeleton>
-        );
+    if (!config || !customer || !checkout || !cart) {
+        throw new Error('Unable to access checkout data')
     }
 
-    private handleSubmit: (values: BillingFormValues) => void = async ({
-        orderComment,
-        ...addressValues
-    }) => {
-        const {
-            updateAddress,
-            updateCheckout,
-            customerMessage,
-            billingAddress,
-            navigateNextStep,
-            onUnhandledError,
-        } = this.props;
-
+    const isInitializing  = isLoadingBillingCountries();
+    const themeV2  = isExperimentEnabled(config.checkoutSettings,
+        'CHECKOUT-7962.update_font_style_on_checkout_page');
```

Comment:
Why are we reading this from setting, shouldn't this be read from the context?

Reply:
I got it wrong. Yes, I should be able to get it from the context.